src/main/java/use_case/leaderboard_as_chart/LeaderboardAsChartInteractor.java

```diff
@@ -0,0 +1,33 @@
+package use_case.leaderboard_as_chart;
+
+import entity.User;
+import frameworks_and_drivers.DataAccess.LeaderboardUserDataAccessObject;
+import use_case.DataAccessException;
+import use_case.leaderboard.LeaderboardType;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class LeaderboardAsChartInteractor implements LeaderboardAsChartInputBoundary {
+
+    private final LeaderboardUserDataAccessObject dao;
+    private final LeaderboardAsChartOutputBoundary presenter;
+
+    public LeaderboardAsChartInteractor(LeaderboardUserDataAccessObject dao, LeaderboardAsChartOutputBoundary presenter) {
+        this.dao = dao;
+        this.presenter = presenter;
+    }
+
+    @Override
+    public void execute() throws DataAccessException {
+        // Get top users for all types
+        Map<LeaderboardType, List<User>> chartData = new HashMap<>();
+        for (LeaderboardType type : LeaderboardType.values()) {
+            chartData.put(type, new ArrayList<>(dao.getTopUsers(50).get(type)));
```

Comment:
Consider passing a parameter `entryCount` instead of using the constant value 50.
```suggestion
            chartData.put(type, new ArrayList<>(dao.getTopUsers(inputData.getLeaderboardEntryCount()).get(type)));
```